enhancements/sig-architecture/6-placements/README.md

```diff
@@ -0,0 +1,523 @@
+// to the placement namespace.
+//
+// Here is how the placement policy combines with other selection methods to determine a matching
+// list of clusters:
+// 1) Clusters are registered with hub as cluster-scoped ManagedClusters;
+// 2) ManagedClusters are organized into cluster-scoped ManagedClusterSets;
+// 3) ManagedClusterSets are bound to workload namespaces;
+// 4) Namespace-scoped Placements specify a slice of ManagedClusterSets which select a working set
+//    of potential ManagedClusters;
+// 5) Then Placements subselect from that working set using label/claim selection.
+//
+// No ManagedClusters will be selected if there is no ManagedClusterSet bound to the placement
+// namespace. User is able to bind a ManagedClusterSet to a namespace by creating a 
+// ManagedClusterSetBinding in the namespace if they have an RBAC rule to CREATE on the virtual
+// subresource of `managedclustersets/bind`.
+//
+// A slice of PlacementDecisions with label placement.open-cluster-management.io={placement name}
+// will be created to represent the cluster selected by this placement.
+// 
+// If a ManagedCluster is selected and added into the PlacementDecisions, other components may apply
+// workload on it; once it is removed from the PlacementDecisions, the workload applied on this
+// ManagedCluster should be evicted accordingly.
+type Placement struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the attributes of Placement.
+	Spec PlacementSpec `json:"spec"`
+
+	// Status represents the current status of the Placement
+	// +optional
+	Status PlacementStatus `json:"status,omitempty"`
+}
+
+// PlacementSpec defines the attributes of Placement.
+// An empty PlacementSpec selects all clusters from the bound ManagedClusterSets to placement namespace.
+type PlacementSpec struct {
+	// Predicates represent a slice of predicates to select clusters. The predicates are ORed.
+	// +optional
+	Predicates []ClusterPredicate `json:"predicates,omitempty"`
+
+	// Affinity represents the scheduling constraints
+	// +optional
+	Affinity *Affinity `json:"affinity,omitempty"`
+}
+
+// ClusterPredicate represents a predicate to select clusters.
+type ClusterPredicate struct {
+	// ClusterSets represent the ManagedClusterSets from which the clusters are selected.
+	// If the slice is empty, clusters will be selected from the ManagedClusterSets bound to the placement
+	// namespace, otherwise clusters will be selected from the intersection of this slice and the
+	// ManagedClusterSets bound to the placement namespace.
+	// +optional
+	ClusterSets []string `json:"clusterSets,omitempty"`
+
+	// NumberOfClusters represents the number of clusters to be selected which meet the
+	// requirements in this predicate.
+	// Nil indicates that NumberOfClusters is not set and there is no bound number.
+	// +optional
+	NumberOfClusters *int32 `json:"numberOfClusters,omitempty"`
```

Comment:
precisely, this should be `maxNumberOfClusters` considering a case that the matching clusters doesnt reach the number

Reply:
the design currently is that the number must be met, otherwise decision will not be created. maxNumberOfClusters would be a different semantics, since we would also  need minNumberOfClusters together.

Reply:
> This field seems like it expresses 'min' semantics. If this field is valued at 5, and one of the clusters is deprovisioned and removed from the set, what happens? I would expect that a new cluster will be selected automatically.

Yes, a new cluster will be selected if possible.

> What happens if this field is valued at 5 but there are only 3 clusters available?

The original thought is that placementdecision will still be created with 3 clusters, and the condition `PlacementSatisfied` in the status of placement will be set to false, while @qiujian16 suggested that no placementdecision will be created in this situation.

Reply:
I updated the doc of this field to make it more clear.